packages/material-ui/src/Alert/Alert.js

```diff
@@ -1,147 +1,213 @@
 import { useThemeVariants } from '@material-ui/styles';
-import withStyles from '../styles/withStyles';
+import experimentalStyled from '../styles/experimentalStyled';
 import { darken, lighten } from '../styles/colorManipulator';
 import capitalize from '../utils/capitalize';
 import Paper from '../Paper';
+import { getAlertUtilityClass } from './alertClasses';
 import IconButton from '../IconButton';
 import SuccessOutlinedIcon from '../internal/svg-icons/SuccessOutlined';
 import ReportProblemOutlinedIcon from '../internal/svg-icons/ReportProblemOutlined';
 import ErrorOutlineIcon from '../internal/svg-icons/ErrorOutline';
 import InfoOutlinedIcon from '../internal/svg-icons/InfoOutlined';
 import CloseIcon from '../internal/svg-icons/Close';
 
-export const styles = (theme) => {
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...styles[`${styleProps.variant}${capitalize(styleProps.color || styleProps.severity)}`],
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { variant, color, severity, classes } = styleProps;
+
+  const slots = {
+    root: ['root', `${variant}${capitalize(color || severity)}`],
+    icon: ['icon'],
+    message: ['message'],
+    action: ['action'],
+  };
+
+  return composeClasses(slots, getAlertUtilityClass, classes);
+};
+
+const AlertRoot = experimentalStyled(
+  Paper,
+  {},
+  {
+    name: 'MuiAlert',
+    slot: 'Root',
+    overridesResolver,
+  },
+)(({ theme, styleProps }) => {
   const getColor = theme.palette.mode === 'light' ? darken : lighten;
   const getBackgroundColor = theme.palette.mode === 'light' ? lighten : darken;
 
   return {
     /* Styles applied to the root element. */
-    root: {
-      ...theme.typography.body2,
-      borderRadius: theme.shape.borderRadius,
-      backgroundColor: 'transparent',
-      display: 'flex',
-      padding: '6px 16px',
-    },
+    ...theme.typography.body2,
+    borderRadius: theme.shape.borderRadius,
+    backgroundColor: 'transparent',
+    display: 'flex',
+    padding: '6px 16px',
     /* Styles applied to the root element if `variant="filled"`. */
     filled: {},
     /* Styles applied to the root element if `variant="outlined"`. */
     outlined: {},
     /* Styles applied to the root element if `variant="standard"`. */
     standard: {},
     /* Styles applied to the root element if `variant="standard"` and `color="success"`. */
-    standardSuccess: {
-      color: getColor(theme.palette.success.main, 0.6),
-      backgroundColor: getBackgroundColor(theme.palette.success.main, 0.9),
-      '& $icon': {
-        color: theme.palette.success.main,
-      },
-    },
+    ...(styleProps.variant === 'standard' &&
+      styleProps.color === 'success' && {
+        color: getColor(theme.palette.success.main, 0.6),
+        backgroundColor: getBackgroundColor(theme.palette.success.main, 0.9),
+        '& $icon': {
```

Comment:
```suggestion
        [`& .${alertClasses.icon}`]: {
```

x all other occurences :) 

See https://gist.github.com/mnajdova/b0562785b7cf1b9d9aeae382d578594b#some-got-ya-s